include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -112,12 +113,15 @@ class pool_memory_resource final
    * @param maximum_pool_size Maximum size, in bytes, that the pool can grow to. Defaults to all
    * of the available memory on the current device.
    */
-  template <typename Upstream2 = Upstream, 
-            cuda::std::enable_if_t<cuda::mr::resource_with<Upstream, cuda::mr::device_accessible>, int> = 0>
+  template <
+    typename Upstream2 = Upstream,
```

Comment:
What is `Upstream2` used for? Can't see any use.

Reply:
Oh that is actually a bug.

In a nutshell you cannot directly remove constructors via [SFINAE](https://godbolt.org/z/33fncrhcK). 
```cpp
template<class T>
struct Bad {
    template<std::enable_if_t<std::is_same_v<T, int>, int> = 0>
    explicit Bad(const int) {}

    explicit Bad(const float) {}
};
```

To circumvent that, one needs to add an indirection via a [defaulted type ](https://godbolt.org/z/oaYqc16h4)
```cpp
template<class T>
struct Good {    
    template<class T2 = T, std::enable_if_t<std::is_same_v<T2, int>, int> = 0>
    explicit Good(const int) {}

    explicit Good(const float) {}
};
```

Ergo that should should be fixed to use `Upstream2` in the constraint